packages/dashboard/src/app/views/myStories/filters/taxonomy/useTaxonomyFilters.js

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+} from '@googleforcreators/react';
+
+/**
+ * Internal dependencies
+ */
+import useApi from '../../../../api/useApi';
+
+export const getFilterPlaceholder = (filter) => {
+  return filter.labels?.allItems || filter.labels?.name || filter.restBase;
+};
+
+export const getFilterAriaLabel = (filter) => {
+  const ariaLable =
+    filter.labels?.singularName || filter.labels?.name || filter.restBase;
+  return `Filter stories by ${ariaLable}`;
+};
+
+export const getFilterSearchResultsLabel = (filter) => {
+  return (
+    filter.labels?.searchItems ||
+    (filter.labels?.name ? `Search ${filter.labels.name}` : 'Search Taxonomy')
```

Comment:
Ditto here about missing `__()` calls, but again, see my below suggestions that make this function here obsolete.

Reply:
Also you `sprintf` here.